crates/env/src/api.rs

```diff
@@ -526,6 +526,28 @@ where
     })
 }
 
+/// Returns the buffer back to the caller of the executed contract.
+///
+/// # Note
+///
+/// This function  stops the execution of the contract immediately.
+pub fn return_value_scoped(return_flags: ReturnFlags, return_value: &[u8]) -> ! {
+    <EnvInstance as OnInstance>::on_instance(|instance| {
+        EnvBackend::return_value_scoped(instance, return_flags, return_value)
+    })
+}
+
+/// Returns the buffer back to the caller of the executed contract.
+///
+/// # Note
+///
+/// This function  stops the execution of the contract immediately.
+pub fn input_scoped(buffer: &mut [u8]) {
```

Comment:
```suggestion
pub fn input_bytes(buffer: &mut [u8]) -> Result<()> {
```

Reply:
Another problem I have with this API is that it does not cover the erraneous case where `buffer` is too small to hold the entire given input.

Reply:
But we used the same schema in the previous implementation. We decoded the whole ink's message variant